lib/state.go

```diff
@@ -67,8 +67,20 @@ type State struct {
 	// TODO: maybe use https://golang.org/pkg/sync/#Pool ?
 	BPool *bpool.BufferPool
 
-	Vu, Iteration int64
-	Tags          map[string]string
+	Vu, VUIDScenario uint64
+	Iteration        int64
+	Tags             map[string]string
+	// These will be assigned on VU activation.
+	// Returns the iteration number of this VU in the current scenario.
+	GetScenarioVUIter func() uint64
+	// Returns the iteration number across all VUs in the current scenario
+	// unique to this single k6 instance.
+	// TODO: Maybe this doesn't belong here but in ScenarioState?
+	GetScenarioLocalVUIter func() uint64
+	// Returns the iteration number across all VUs in the current scenario
+	// unique globally across k6 instances (taking into account execution
+	// segments).
+	GetScenarioGlobalVUIter func() uint64
```

Comment:
These 2 aren't relevant to a single VU/State, but I added them here since it was simpler to call them from the same place where we increment the other iterations. They would probably be better off as part of `ScenarioState` instead, but that would imply exposing a `ScenarioState.IncrIteration()` method, which we wanted to avoid for the VU `State`, so it's the same situation.

This doesn't bother me too much, but let me know if it should be changed.